### PR TITLE
Add --all flag to podman kill

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: cmd/podman/kill.go [hunk 1]
```diff
@@ -37,12 +42,18 @@ var (
 // killCmd kills one or more containers with a signal
 func killCmd(c *cli.Context) error {
 	args := c.Args()
-	if len(args) == 0 && !c.Bool("latest") {
+	if (!c.Bool("all") && !c.Bool("latest")) && len(args) == 0 {
 		return errors.Errorf("specify one or more containers to kill")
 	}
-	if len(args) > 0 && c.Bool("latest") {
+	if (c.Bool("all") || c.Bool("latest")) && len(args) > 0 {
 		return errors.Errorf("you cannot specific any containers to kill with --latest")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/kill.go]: Maybe Saturday brain, but isn't this going to trigger if you have just `podman kill --all`?  I think you can drop the all check here.

Comment (TomSweeneyRedHat) [reply cmd/podman/kill.go]: Line 49, not yours, but "specific" to "specify" please.

**Comment by mheon** [comment]
I think `kill --all` should mean "kill all processes in the container" instead, unless we have another good name for that flag?

**Comment by baude** [comment]
> I think `kill --all` should mean "kill all processes in the container" instead, unless we have another good name for that flag?

isnt that called stop?  --all does have the precedent of being used elsewheres soley for "all containers"

**Comment by mheon** [comment]
Stop doesn't hit anything except container init, same as kill. All
processes will hit other processes including exec sessions.

On Sep 29, 2018 12:03, "Brent Baude" <notifications@github.com> wrote:

I think kill --all should mean "kill all processes in the container"
instead, unless we have another good name for that flag?

isnt that called stop? --all does have the precedent of being used
elsewheres soley for "all containers"

—
You are receiving this because your review was requested.

Reply to this email directly, view it on GitHub
<https://github.com/containers/libpod/pull/1568#issuecomment-425655839>,
or mute
the thread

**Comment by rhatdan** [comment]
I dissagree.  I think kill all means kill all containers.  That is what I was trying to do when I  noticed it was not working.  I was giving demos of podman with an updated version of @umohnani8 talk, and I wanted to kill all of the containers that were running without waiting for the stop signal to happen.

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by vrothberg** [comment]
Just two minor things:

- I think that "**Kill** all running containers" is more comprehensible for users than "**Signal** all running containers". Technically, a kill signal is being sent but I am not sure tha all users are familiar with signal handling and might be confused.

- I would like to add a test for this flag to avoid potential future regressions.

**Comment by mheon** [comment]
I still want an all processes flag, but you're right that it would be confusing to repurpose --all when it means other things elsewhere. Would like to see a test like @vrothberg said though.

**Comment by mheon** [comment]
LGTM

**Pull request merged**